### PR TITLE
use seed proxy settings for the seed deployed webhook

### PR DESCRIPTION
**Pull request by mate4st**
**What this PR does / why we need it**:

This should have been fixed in https://github.com/kubermatic/kubermatic/pull/11561 but I used the wrong proxy configuration for the seed webhook.

The webhook can be deployed in master and seed. In a dedicated Seed we need to use the seed proxy configuration
The proxy in Kubermatic configuration should only be used for master cluster deployments. 

**What type of PR is this?**

/kind bug
/kind regression

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Use seed proxy configuration for seed deployed webhook.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by WeirdMachine** [comment]
> ```
> {"level":"info","time":"2023-03-17T10:24:43.256Z","caller":"runtime/panic.go:884","msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"kkp-master-operator","object":{"name":"e2e","namespace":"kubermatic"},"namespace":"kubermatic","name":"e2e","reconcileID":"d0c1c745-be37-4e3a-97b0-dc83bb1df3a1"}
> panic: runtime error: invalid memory address or nil pointer dereference [recovered]
> 	panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x288 pc=0x1865c82]
> 
> goroutine 491 [running]:
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:119 +0x1fa
> panic({0x1ae04a0, 0x2ffe6b0})
> 	runtime/panic.go:884 +0x213
> k8c.io/kubermatic/v2/pkg/controller/operator/common.WebhookDeploymentReconciler.func1.1(0xc000311900)
> 	k8c.io/kubermatic/v2/pkg/controller/operator/common/webhook.go:219 +0x8c2
> k8c.io/reconciler/pkg/reconciling.DefaultDeployment.func1(0xc000311900?)
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/defaults.go:110 +0xa2
> k8c.io/reconciler/pkg/reconciling.DeploymentObjectWrapper.func1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/zz_generated_reconcile.go:493 +0x46
> k8c.io/reconciler/pkg/reconciling.CreateWithNamespace.func1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/ensure.go:46 +0x3d
> k8c.io/reconciler/pkg/reconciling.CreateWithName.func1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/ensure.go:57 +0x3d
> k8c.io/kubermatic/v2/pkg/controller/operator/common.OwnershipModifierFactory.func1.1({0x214b260?, 0xc000311900?})
> 	k8c.io/kubermatic/v2/pkg/controller/operator/common/util.go:105 +0x5b
> k8c.io/kubermatic/v2/pkg/controller/operator/common.VolumeRevisionLabelsModifierFactory.func1.1({0x214b260?, 0xc000311900?})
> 	k8c.io/kubermatic/v2/pkg/controller/operator/common/util.go:154 +0x64
> k8c.io/reconciler/pkg/reconciling.ImagePullSecretsWrapper.func1.1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/wrapper.go:52 +0x5b
> k8c.io/reconciler/pkg/reconciling.CreateWithNamespace.func1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/ensure.go:46 +0x3d
> k8c.io/reconciler/pkg/reconciling.CreateWithName.func1({0x214b260?, 0xc000311900?})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/ensure.go:57 +0x3d
> k8c.io/reconciler/pkg/reconciling.EnsureNamedObject({0x21304e8, 0xc000166780}, {{0xc000c53074?, 0x20?}, {0x1d7f3a8?, 0x7f085860c3c8?}}, 0x20?, {0x2139bf8, 0xc0006c48a0}, {0x214b260, ...}, ...)
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/ensure.go:122 +0x602
> k8c.io/reconciler/pkg/reconciling.ReconcileDeployments({0x21304e8, 0xc000166780}, {0xc0009a39e0, 0x2, 0x6?}, {0xc000c53074, 0xa}, {0x2139bf8, 0xc0006c48a0}, {0xc0006c6e20, ...})
> 	k8c.io/reconciler@v0.3.1/pkg/reconciling/zz_generated_reconcile.go:512 +0x18b
> k8c.io/kubermatic/v2/pkg/controller/operator/master.(*Reconciler).reconcileDeployments(0xc000562780, {0x21304e8, 0xc000166780}, 0xc000980000, 0x214ae00?)
> 	k8c.io/kubermatic/v2/pkg/controller/operator/master/reconciler.go:352 +0x509
> k8c.io/kubermatic/v2/pkg/controller/operator/master.(*Reconciler).reconcile(0xc000314198?, {0x21304e8, 0xc000166780}, 0xc0004e0300, 0x21304e8?)
> 	k8c.io/kubermatic/v2/pkg/controller/operator/master/reconciler.go:136 +0x32d
> k8c.io/kubermatic/v2/pkg/controller/operator/master.(*Reconciler).Reconcile(0xc000562780, {0x21304e8, 0xc000166780}, {{{0xc000c53074, 0xa}, {0xc000c53070, 0x3}}})
> 	k8c.io/kubermatic/v2/pkg/controller/operator/master/reconciler.go:80 +0x2b4
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x21304e8?, {0x21304e8?, 0xc000166780?}, {{{0xc000c53074?, 0x1a4d760?}, {0xc000c53070?, 0x0?}}})
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:122 +0xc8
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0005628c0, {0x2130440, 0xc000051630}, {0x1b60a60?, 0xc0009044c0?})
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:323 +0x35f
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0005628c0, {0x2130440, 0xc000051630})
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:274 +0x1d9
> sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:235 +0x85
> created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
> 	sigs.k8s.io/controller-runtime@v0.14.4/pkg/internal/controller/controller.go:231 +0x587
> ```

yeah this turned out to be a little bit more complicated than I thought it is :D

File: pkg/controller/operator/common/webhook.go [hunk 1]
```diff
@@ -197,36 +197,45 @@ func WebhookDeploymentReconciler(cfg *kubermaticv1.KubermaticConfiguration, vers
...
-				} else if d != nil && len(d.Spec.Template.Spec.Containers) > 0 {
-					// check if the old Deployment had a seed env var
-					for _, e := range d.Spec.Template.Spec.Containers[0].Env {
-						if e.Name == seedNameEnvVariable {
-							seedName = e.Value
-							break
-						}
+			var envVars []corev1.EnvVar
+
+			// The information weather a Seed is present or not is stored in the seedNameEnvVariable.
```
Comment (xrstf) [inline pkg/controller/operator/common/webhook.go]: weather != whether

**Review by xrstf — APPROVED** [review]
/approve

**Comment by embik** [comment]
@WeirdMachine this still needs a backport, right?

**Comment by embik** [comment]
@WeirdMachine reminder this needs a backport to 2.20.

**Pull request merged**